Add GraalVM Native Image docs (#302)
cap-js/docs: java/development/index.md

### Spring Features

Beside the common Spring features such as dependency injection and a sophisticated test framework, the following features are available in Spring CAP applications:

* CDS event handlers within custom Spring beans are automatically registered at startup.
* Full integration into Spring transaction management (`@Transactional` is supported).

> * A number of CAP Java SDK interfaces are exposed as [Spring beans](#exposed-beans) and are available in Spring application contexts such as technical services, the `CdsModel`, or the `UserInfo` in current request scope.

Author: There is only one spring application context in an application.

```suggestion
* A number of CAP Java SDK interfaces are exposed as [Spring beans](#exposed-beans) and are available in the Spring application context such as technical services, the `CdsModel`, or the `UserInfo` in current request scope.
```

1. The GraalVM Native Image build analyzes your application from the `main` entry point. Only the code that is reachable through static analysis is included into the native image. This means that the full classpath needs to be known and available already at build time.

2. Dynamic elements of your code, such as usage of reflection, JDK proxies, or resources need to be registered with the GraalVM Native Image build. You can learn more about this in the [GraalVM Native Image documentation](https://www.graalvm.org/latest/reference-manual/native-image/metadata/).

    ::: tip
    Many runtime hints for reflection, JDK proxy usage, and resources are contributed automatically to the Native Image build.
    This includes 
    - Required reflection for event handler classes defined in application code

>     - The usage of a required JDK proxy for interfaces generated from the application's CDS model by the CDS Maven Plugin.

Author: ```suggestion
    - The usage of JDK proxies for interfaces generated from the application's CDS model by the CDS Maven Plugin.
```

If the Spring Boot Devtools configuration of your CAP Java application defines a [trigger file](https://docs.spring.io/spring-boot/docs/current/reference/html/using.html#using.devtools.restart.triggerfile), the `auto-build` can detect this and touch the trigger file in case of any file change. The same applies to the `watch` goal.
:::

#### Local Development for Multitenant Applications

> With the streamlined MTX, you can run your multitenant application locally along with the MTX sidecar and use SQLite as the database. See [the _Deploy as SaaS_ guide](../../guides/deployment/as-saas#local-mtx) for more information.

Reviewer: We could even add the toggle.
```suggestion
With the streamlined MTX, you can run your multitenant application locally along with the MTX sidecar and use SQLite as the database. See [the _Deploy as SaaS_ guide](../../guides/deployment/as-saas?impl-variant=java#local-mtx) for more information.
```
